src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -566,20 +563,22 @@ def test_copy():
     op2 = Literal("2", INTEGER_TYPE)
     call = Call.create(RoutineSymbol("name"), [("name1", op1), ("name2", op2)])
     # consistent call
-    call_copy = call.copy()
-    assert call._argument_names[0] == (id(call.children[0]), "name1")
-    assert call._argument_names[1] == (id(call.children[1]), "name2")
-    assert call_copy._argument_names[0] == (id(call_copy.children[0]), "name1")
-    assert call_copy._argument_names[1] == (id(call_copy.children[1]), "name2")
-    assert call._argument_names != call_copy._argument_names
-
-    call.children = [op2.detach(), op1.detach()]
-    assert call._argument_names[0] != (id(call.children[0]), "name2")
-    assert call._argument_names[1] != (id(call.children[1]), "name1")
+    call2 = call.copy()
+    assert call._argument_names[0] == (id(call.arguments[0]), "name1")
+    assert call._argument_names[1] == (id(call.arguments[1]), "name2")
+    assert call2._argument_names[0] == (id(call2.arguments[0]), "name1")
+    assert call2._argument_names[1] == (id(call2.arguments[1]), "name2")
+    assert call._argument_names != call2._argument_names
+
+    # Swap position of arguments
+    call.children.extend([op2.detach(), op1.detach()])
+
+    assert call._argument_names[0] != (id(call.arguments[0]), "name2")
+    assert call._argument_names[1] != (id(call.arguments[1]), "name1")
     # inconsistent call
```

Comment:
Not yours but do you understand what "inconsistent call" means because I don't? :-)

Reply:
Ah yeah, it means the internal state of the _argument_name tuple is inconsistent and needs to be reconciled before the copy happens. Updated the comments.

Reply:
Aha - thanks!